### ISSUE TITLE
[prefer-readonly-parameter-types] extending ReadonlyArray gives false positive

### ISSUE THREAD

**Issue by ckknight** [typescript-eslint/typescript-eslint#1789]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-readonly-parameter-types": "error"
  }
}
```

```TS
interface CustomArrayType extends ReadonlyArray<string> {
  readonly prop: string;
}
function custom1(arg: CustomArrayType) {}
```

**Expected Result**
No error

**Actual Result**
Parameter should be a read only type

**Additional Info**
This is the example within https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-readonly-parameter-types.md

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.25.0` |
| `@typescript-eslint/parser`        | `2.25.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
sorry, the examples are incorrect, the tests are correct:

https://github.com/typescript-eslint/typescript-eslint/blob/258fdd059526175431dbfe60dc88e0cc9d99c33e/packages/eslint-plugin/tests/rules/prefer-readonly-parameter-types.test.ts#L121-L126

Because the type extends an array type, it is no longer classified as an array by TS.
Which means it's just treated as a plain old JS object.
